admiral/pkg/controller/admiral/dependency_test.go

```diff
@@ -20,3 +24,88 @@ func TestNewDependencyController(t *testing.T) {
 		t.Errorf("Dependency controller should never be nil without an error thrown")
 	}
 }
+
+func TestGetDependencies(t *testing.T) {
+	stop := make(chan struct{})
+	handler := test.MockDependencyHandler{}
+
+	dependencyController, err := NewDependencyController(stop, &handler, "../../test/resources/admins@fake-cluster.k8s.local", "ns", time.Duration(1000))
+
+	if err != nil {
+		t.Errorf("Unexpected err %v", err)
+	}
+
+	if dependencyController == nil {
+		t.Errorf("Dependency controller should never be nil without an error thrown")
+	}
+
+	deps, _  := dependencyController.GetDependencies()
+
+	if deps != nil {
+		t.Errorf("deps is not nil")
+	}
+
+	dep := model.Dependency{IdentityLabel: "identity", Destinations:[]string{"greeting", "payments", "newservice"}, Source: "webapp"}
+	depObj := makeK8sDependencyObj("mydep", "namespace1", dep)
+
+	dependencyController.DepCrdClient.AdmiralV1().Dependencies("namespace1").Create(depObj);
+
+	deps, _  = dependencyController.GetDependencies()
+
+	if deps == nil || len(deps) == 0{
```

Comment:
Can we assert that the objects match, rather than just non-empty?

Reply:
Done